packages/panzer/mini-em/src/solvers/MiniEM_DiscreteCurl.hpp

```diff
@@ -0,0 +1,482 @@
+#include "Panzer_LOCPair_GlobalEvaluationData.hpp"
+#include "Panzer_IntrepidOrientation.hpp"
+#include "Panzer_IntrepidBasisFactory.hpp"
+#include "Intrepid2_OrientationTools.hpp"
+#include "Intrepid2_LagrangianInterpolation.hpp"
+#include "Thyra_EpetraThyraWrappers.hpp"
+
+class CurlRequestCallback : public Teko::RequestCallback<Teko::LinearOp> {
+private:
+
+  Teko::LinearOp curl_;
+
+public:
+
+  CurlRequestCallback(const Teko::LinearOp & curl)
+     : curl_(curl) {};
+
+  bool handlesRequest(const Teko::RequestMesg & rm)
+  {
+    std::string name = rm.getName();
+
+    return (name=="Discrete Curl");
+  };
+
+  Teko::LinearOp request(const Teko::RequestMesg & rm)
```

Comment:
So apparently Teko has generalized [Greenspun's Tenth Rule](https://en.wikipedia.org/wiki/Greenspun%27s_tenth_rule) to Smalltalk ;-)

Reply:
Indeed, though its far simpler...you could read the Teko paper it explains it very clearly.